code/modules/mob/living/basic/icemoon/ice_whelp/ice_whelp_abilities.dm

```diff
@@ -12,7 +12,8 @@
 /datum/action/cooldown/mob_cooldown/ice_breath/Activate(atom/target_atom)
 	var/turf/target_fire_turf = get_ranged_target_turf_direct(owner, target_atom, fire_range)
```

Comment:
maybe add should_not_sleep

Reply:
To be clear do you think this should go here, in `action/cooldown/Activate`, or in `action/Trigger`?
Putting `should_not_sleep` in `mob_cooldown/fire_breath/proc/fire_line` when I already know it sleeps in order to function seems slightly self-defeating unless I am misunderstanding something

Reply:
`SHOULD_NOT_SLEEP` should be used wherever you would put `SIGNAL_HANDLER` (as in the proc that actually reacts to an input occuring, and that will then dispatch other procs "asynchronously"), so I suppose that would have to be on the base definition `Activate()` since that is the key proc that responds to the input (a player pushing the button) and is the proc that will triage the type of input it recieved/context and dispatch any resultant procs

i know that doesn't make any sense but that's how i think of it. i'm pretty sure the whole SHOULD_NOT_SLEEP thing might add a shitload more langserver-time errors because there's bound to be a lot of stuff that sleeps in this so we might have to resolve that at a later date. You would get a similar effect if you put it on Trigger() too, since Trigger invokes Activate (iirc), and SHOULD_NOT_SLEEP will assess any proc that is further invoked. langserver stuff idfk glarg whatever makes you happy

Reply:
I have decided that we do _not_ want `DO_NOT_SLEEP` to be a requirement on abilities triggering in general.
Enough of them use input or do_afters in a way which seems pretty valid.
We just need to be careful when doing something in an ability which sleeps and not checking if we were destroyed afterwards before continuing to act.